contracts/instance/base/ComponentServiceBase.sol

```diff
@@ -43,72 +28,28 @@ abstract contract ComponentServiceBase is Service {
         _;
     }
 
-    function _initializeService(
-        address registry, 
-        address initialOwner
-    )
-        internal
-        override
-    {
-        super._initializeService(registry, initialOwner);
-        _objectTypeToExpectedRole[PRODUCT()] = PRODUCT_OWNER_ROLE();
-        _objectTypeToExpectedRole[POOL()] = POOL_OWNER_ROLE();
-        _objectTypeToExpectedRole[DISTRIBUTION()] = DISTRIBUTION_OWNER_ROLE();
-        _objectTypeToExpectedRole[ORACLE()] = ORACLE_OWNER_ROLE();
+    // view functions
+
+    function getRegistryService() public view virtual returns (IRegistryService) {
+        address service = getRegistry().getServiceAddress("RegistryService", getMajorVersion());
+        return IRegistryService(service);
     }
 
     function getInstanceService() public view returns (InstanceService) {
         return InstanceService(getRegistry().getServiceAddress("InstanceService", getMajorVersion()));
     }
 
-    function register(address componentAddress) 
```

Comment:
Why did you move the `register` function back into each service? we wanted to have it centralized to ensure the common functions - especially the component owner role checks and the assignment which role is required for which service -  are always called.

Reply:
register() is special to product, pool and distribution service, while policy and bundle service have createApplication() and creatBundle()